module/data/item/templates/class.mjs

```diff
@@ -165,12 +165,31 @@ export default class ClassTemplate extends ItemDataModel.mixin(
       nextLevelData.abilities.special.map( ability => fromUuid( ability ) )
     );
 
+
+
     const effects = Array.from( nextLevelData.effects );
 
     await this.parentActor.createEmbeddedDocuments( "Item", [ ...freeAbilityData, ...specialAbilityData ] );
     await this.parentActor.createEmbeddedDocuments( "ActiveEffect", effects );
     // TODO: activate permanent effects immediately
 
+    // increase resource step of the discipline
+
+    const highestClass = this.parentActor.items.filter(
+      item => item.type === "discipline"
+    ).sort(     // sort descending by circle/rank
+      ( a, b ) => a.system.level > b.system.level ? -1 : 1
+    )[0];
+    
+    if ( this.parent.type === "discipline" && this.parent === highestClass ) {
```

Comment:
Object comparison is not obvious. I'd rather compare their IDs to be sure

Reply:
good point, did it.